src/resources/ble_esp32.cc

```diff
@@ -139,53 +156,105 @@ class BLEResourceGroup : public ResourceGroup {
     ble_pool.put(_id);
   }
 
-  GAPResource* gap() { return _gap; }
-  void set_gap(GAPResource* gap) { _gap = gap; }
+  GAPResource *gap() { return _gap; }
 
-  uint32_t on_event(Resource* resource, word data, uint32_t state);
+  void set_gap(GAPResource *gap) { _gap = gap; }
 
-  bool is_advertising() { return _advertising; }
-  void set_advertising(bool value) { _advertising = value; }
+  BLEServerConfigGroup *server_config() { return _server_config; }
 
-  bool is_scanning() { return _scanning; }
-  void set_scanning(bool value) { _scanning = value; }
+  void set_server_config(BLEServerConfigGroup *server_config) { _server_config = server_config; }
 
-  uint16 connection_handle() const {
-    Locker locker(_mutex);
-    return _connection_handle;
-  }
+  uint32_t on_event(Resource *resource, word data, uint32_t state) override;
 
-  void clear_connection_handle() {
-    Locker locker(_mutex);
-    _connection_handle = kInvalidHandle;
-  }
+  int init_server();
+
+  static bool is_advertising() { return ble_gap_adv_active(); }
 
-  BLEDiscovery* next() {
+  static bool is_scanning() { return ble_gap_disc_active(); }
+
+  BLEDiscovery *next() {
     Locker locker(_mutex);
     return _discoveries.first();
   }
 
   bool remove_next() {
     Locker locker(_mutex);
-    BLEDiscovery* next = _discoveries.first();
+    BLEDiscovery *next = _discoveries.first();
     if (!next) return false;
     _discoveries.remove_first();
     delete next;
     return true;
   }
 
- private:
+private:
   int _id;
-  Mutex* _mutex;
+  Mutex *_mutex;
   DiscoveriesFIFO _discoveries;
-  GAPResource* _gap = null;
-  uint16 _connection_handle = kInvalidHandle;
-  bool _advertising = false;
-  bool _scanning = false;
+  GAPResource *_gap = null;
+  BLEServerConfigGroup *_server_config;
 };
 
-uint32_t BLEResourceGroup::on_event(Resource* resource, word data, uint32_t state) {
-  struct ble_gap_event* event = reinterpret_cast<struct ble_gap_event*>(data);
+
+class BLEServerConfigGroup : public ResourceGroup, public Thread {
+public:
+  TAG(BLEServerConfigGroup);
+
+  explicit BLEServerConfigGroup(Process *process, EventSource* event_source) :
+      ResourceGroup(process, event_source), Thread("toit.ble"),
+      _gatt_services(null) {}
+
+  ~BLEServerConfigGroup() override {
```

Comment:
Can this be protected? We need to through `tear_down`, so I don't think this should be exposed.